Refactor usage of MixedRealityExtensionApp; fix build path error with spaces

influnza opened the pull request:
This change only refactors, no change to functionality. 
Removed IConnectionInternal that is not used.
Made several classes public: this will help integrating the part that handles the connection into another 3D engine potentially.
Replaced MixedRealityExtensionApp with its interface, where possible.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/mixed-reality-extension-unity?pullRequest=71) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/mixed-reality-extension-unity?pullRequest=71) before we can accept your contribution.<br/><br/>:x: influnza [sign now](https://cla.opensource.microsoft.com/Microsoft/mixed-reality-extension-unity?pullRequest=71)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/mixed-reality-extension-unity?pullRequest=71) it.</sub>

influnza commented:
> (As discussed IRL) Batch file changes are good, but it doesn't make sense to export additional symbols until such time we split the DLL into Unity and common implementations.

Reverted the change to visibility. Is the change to interfaces acceptable?

eanders-MS commented:
@tombuMS what are your thoughts? Is this a good change?

tombuMS commented on MREUnityRuntime/MREUnityRuntimeLib/App/IMixedRealityExtensionApp.cs:
```diff
@@ -73,6 +74,11 @@ public interface IMixedRealityExtensionApp
+        /// <summary>
+        /// Current connection to the application server
+        /// </summary>
+        IConnection Connection { get; }
```
I don't think we should be exposing this to the host app.  The connection is supposed to be form internal use within the runtime.

tombuMS commented on MREUnityRuntime/MREUnityRuntimeLib/App/IMixedRealityExtensionApp.cs:
```diff
@@ -133,5 +139,16 @@ public interface IMixedRealityExtensionApp
+        /// Send the event, using the current protocol
+        /// </summary>
+        /// <param name="update"></param>
+        void SendEvent(Payload update);
```
Sending events and working with payloads should not be exposed to host app implementations

tombuMS commented on MREUnityRuntime/MREUnityRuntimeLib/App/IMixedRealityExtensionApp.cs:
```diff
@@ -133,5 +139,16 @@ public interface IMixedRealityExtensionApp
+        /// <summary>
+        /// Send the event, using the current protocol, with provide response ID
+        /// </summary>
+        void SendEvent(Payload update, string _responseId);
```
Same as above. This is an internal detail to the runtime that shouldn't be touched by host app integrations.

tombuMS commented:
> @tombuMS what are your thoughts? Is this a good change?

I don't think this is a change we want to take.  The purpose of the IMixedRealityExtensionApp interface is to expose only the subset of functionality that we want the host app to be able to override or needs to be able to call in to.  With that purpose in mind, we use the non-interface concrete class in our internal code so that we have access to the internal functionality that should remain only in the runtime, as it s not needed by the host app integrations.  We can consider exposing things in the interface on a case basis, but all up this change exposes much more than it should I believe.

influnza commented:
Removed the change to app interface.

eanders-MS commented:
Thank you @influnza.
